File: korepetynder-spa/src/app/settings/models/requests/studentLessonRequest.ts
export class StudentLessonRequest {
  preferredCostMinimum: number;
  preferredCostMaximum: number;
  frequency: number | null;
  subjectId: number;
  levelsIds: number[];
  languagesIds: number[];

  constructor(minCost: number, maxCost: number, frequency: number | null, subjectId: number,
    levelsIds: number[], languagesIds: number[]) {
    this.preferredCostMinimum = minCost;
    this.preferredCostMaximum = maxCost;
    this.frequency = frequency;
    this.subjectId = subjectId;
    this.levelsIds = levelsIds;
    this.languagesIds = languagesIds;
  }
}

File: Korepetynder.Contracts/Requests/Students/StudentLessonRequest.cs
namespace Korepetynder.Contracts.Requests.Students
{
    public class StudentLessonRequest
    {
        public int? Frequency { get; set; }
        public int SubjectId { get; set; }
        public int PreferredCostMinimum { get; set; }
        public int PreferredCostMaximum { get; set; }
        public IEnumerable<int> LevelsIds { get; set; }
        public IEnumerable<int> LanguagesIds { get; set; }

        public StudentLessonRequest(int? frequency, int subjectId, IEnumerable<int> levelsIds, IEnumerable<int> languagesIds, int preferredCostMinimum, int preferredCostMaximum)
        {
            PreferredCostMinimum = preferredCostMinimum;
            PreferredCostMaximum = preferredCostMaximum;
            Frequency = frequency;
            SubjectId = subjectId;
            LevelsIds = levelsIds;
            LanguagesIds = languagesIds;
        }

    }
}
